commands/bake.go

```diff
@@ -238,26 +207,23 @@ func runBake(ctx context.Context, dockerCli command.Cli, targets []string, in ba
 		return err
 	}
 
-	def := struct {
+	dtdef, err := json.MarshalIndent(struct {
```

Comment:
Passing this encoded JSON around seems weird. I think we can just call encode in SaveLocalState.